docker.md

```diff
@@ -0,0 +1,120 @@
+# PMO Docker
+
+Wrapping PMO in Docker.
+
+Sections:
+- [Prerequisites](#prerequisites)
+- [Dockerfiles](#dockerfiles)
+- [Building Docker](#building-docker)
+- [Running the app](#running-the-app)
+
+## Prerequisites
+
+- Docker
+
+> **Note:** This guide so far assumes running on a Mac due to Docker cross-container communication.
+> Non-Mac executions will be covered in a future guide.
+
+## Dockerfiles
+
+We'll specify 3 `Dockerfile`s:
+
+- `Dockerfile` - UI
+- `Dockerfile-api` - API
+- `Dockerfile-ci` - CI (tests, etc)
+
+### Dockerfile
+
+The main dockerfile will be used to run the UI application.
+
+`Dockerfile`
+```
+FROM node:10.0
+
+# Create app directory
+WORKDIR /usr/src/app
+
+# get files
+COPY . .
+
+# install dependencies
+RUN npm install donejs -g
+RUN npm install
+
+# If you are building your code for production
+# RUN donejs build
+
+# expose your ports
+EXPOSE 8080
+
+# start it up
+CMD [ "donejs", "develop-docker-mac" ]
+# If you are running production, use
+# ENV NODE_ENV production
+# CMD [ "npm", "start" ]
+```
+
+> **Note:** Notice the `develop-docker-mac` instead of `develop`.  This is because the UI container needs to know how to communicate with the API container.
+> With a Mac, there is a `http://docker.for.mac.localhost` which allows access to another container.  See `scripts["develop-docker-mac"]` in `package.json` for more details.
```

Comment:
What's wrong w/mapping to host ports? Does docker not work w/localhost?

Reply:
localhost from within a container gives you the container's network not docker's network.  there's a [way to define a network](https://docs.docker.com/network/bridge/), but I haven't had time to play with it

Reply:
docker.for.mac.localhost give you docker's network which allows you to access other contianers' ports

Reply:
Hmm, I just I'm just familiar with `-p containerport:hostport` (I think that's how it goes). Is that the same as `--publish containerport:hostport` and `EXPOSE ...`?

Won't that expose a port from the container to the host?

Reply:
not sure.  I haven't tinkered with it, but there's definitely a way.  From the little reading I did, it seems like user-defined networks would be a good solution, so either that or a custom publishing/exposing thing would probably be the way to go.

All that should be done inside of the `./scripts-docker/*` scripts, though, so the outer footprint shouldn't change.

Reply:
This has been resolved with `docker network create`.  see `./scripts/run`